nistats/datasets.py

```diff
@@ -93,46 +104,88 @@ def fetch_openfmri_dataset(dataset_name='ds000001', dataset_revision=None,
-    files = _fetch_file(openfmri_api, data_dir)
-    json_api = json.load(open(files, 'r'))
-
-    dataset_url_set = []
-    for i in range(len(json_api)):
-        # We look for the desired dataset in the json api file
-        if dataset_name == json_api[i]['accession_number']:
-            # Now we look for the desired revision or the last one
-            if not dataset_revision:
-                revision = json_api[i]['revision_set']
-                if revision:
-                    dataset_revision = revision[-1]['revision_number']
-            # After selecting the revision we download all its files
-            link_set = json_api[i]['link_set']
-            for link in link_set:
-                revision = link['revision']
-                if revision == dataset_revision:
-                    dataset_url_set.append(link['url'])
-            # If revision is specified but no file is found there is an issue
-            if dataset_revision and not dataset_url_set:
-                Exception('No files found for revision %s' % dataset_revision)
-            break
-
-    if not dataset_url_set:
-        raise ValueError('dataset %s not found' % dataset_name)
+
+    # First we download the url list from the uncompressed dataset version
+    url_file = os.path.join(data_dir, 'urls.json')
+    urls = []
+    if not os.path.exists(url_file):
+
+        def get_url(endpoint_url, bucket_name, file_key):
+            return '{}/{}/{}'.format(endpoint_url, bucket_name, file_key)
+
+        resource = boto3.resource('s3')
+        resource.meta.client.meta.events.register('choose-signer.s3.*',
+                                                  disable_signing)
+        bucket = resource.Bucket('openneuro')
+
+        for obj in bucket.objects.filter(Prefix=data_prefix):
+            # get url of files (keys of directories end with '/')
+            if obj.key[-1] != '/':
+                urls.append(
+                    get_url(bucket.meta.client.meta.endpoint_url,
+                            bucket.name, obj.key))
+
+        with open(url_file, 'w') as json_file:
+            json.dump(urls, json_file)
     else:
-        # The files_spec needed for _fetch_files
-        files_spec = []
-        for dat_url in dataset_url_set:
-            target_file = os.path.basename(dat_url)
-            url = dat_url
-            files_spec.append((target_file, url, {}))
-        # download the files
-        downloaded_files = _fetch_files(data_dir, files_spec, resume=True,
-                                        verbose=verbose)
-    return data_dir, downloaded_files
+        with open(url_file, 'r') as json_file:
+            urls = json.load(json_file)
+
+    # We apply filters to the urls
+    for exclusion in exclusion_filters:
+        urls = [url for url in urls if not re.match(exclusion, url)]
+    for inclusion in inclusion_filters:
+        urls = [url for url in urls if re.match(inclusion, url)]
+
```

Comment:
I reiterate that with these conventions, exclusion and inclusion filter play logically the same role. Not a big deal though.